include/net/net_if.h

```diff
@@ -1154,6 +1160,28 @@ void net_if_ipv6_prefix_unset_timer(struct net_if_ipv6_prefix *prefix);
  */
 bool net_if_ipv6_addr_onlink(struct net_if **iface, struct in6_addr *addr);
 
+/**
+ * @brief Get the IPv6 address of the given router
+ * @param router a network router
+ *
+ * @return pointer to the IPv6 address, or NULL if none
+ */
+#if defined(CONFIG_NET_IPV6)
+static inline struct in6_addr *net_if_router_ipv6(struct net_if_router *router)
+{
+	return &router->address.in6_addr;
+}
+#else
+static inline struct in6_addr *net_if_router_ipv6(struct net_if_router *router)
+{
+	static struct in6_addr addr;
```

Comment:
Return `net_ipv6_unspecified_address()` here so we save some bytes.

Reply:
good point

Reply:
actually, this will be nuked out by the compiler (that's being made so code using the function can be enclosed in IS_ENABLED() )